### PR TITLE
Add CONTRIBUTING.md and document auto-reload

### PR DESCRIPTION
**Pull request by viennem**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+## Development
+
+### How to test your plugin in leto-modelizer
+
+When you are working on your plugin, you can choose to write your unit tests first to test your code (test-driven development).
+
+Alternatively, if you want to test your plugin in leto-modelizer, it can be tedious to rebuild and reinstall your plugin every time you edit its code. Thankfully, there is a trick to auto-reload the Quazar dev server every time you edit your plugin's source code. Here is how to do it:
+
+1. Make sure you cloned the [leto-modelizer](https://github.com/ditrit/leto-modelizer) and your plugin's repos.
```
Comment (Zorin95670) [inline CONTRIBUTING.md]: repository

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+   * Instead of: `import MyPluginDrawer from 'src/draw/MyPluginDrawer';`
+3. In `leto-modelizer/src/plugins/index.js`, import your plugin with a relative import:
+   ```js
+   import myPlugin from '../../../my-plugin/src'
+
+   export default {
+     myPlugin,
+   };
+   ```
+4. In `leto-modelizer`, run `npm run dev`.
```
Comment (Zorin95670) [inline CONTRIBUTING.md:62]: Miss the edition of `package.json` with :
```
"exports": {
    ".": {
      "development": "./src",
      "default": "./dist/my-plugin.js"
    }
  },
```

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+- Improve the version number in `package.json`, `package-lock.json` and `changelog.md`.
+- Verify the content of the `changelog.md`.
+- Commit the modifications with the label `Release version X.Y.Z`.
+- Create a pull request on github for this branch into `main`.
+- Once the pull request validated and merged, tag the `main` branch with `vX.Y.Z`
+- After the tag is pushed, make the release on the tag in GitHub
+
+### Git: Default branch
+
+The default branch is main. Direct commit on it is forbidden. The only way to update the application is through pull request.
```
Comment (justinehell) [inline CONTRIBUTING.md]: The default branch is `main`.

**Review by justinehell — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (justinehell) [reply CONTRIBUTING.md]: and your plugin repositories.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+### test:coverage
+
+Run all the unit tests and generate coverage report of the unit tests for sonar.
+
+## Development
+
+### How to test your plugin in leto-modelizer
+
+When you are working on your plugin, you can choose to write your unit tests first to test your code (test-driven development).
```
Comment (k-y0u) [inline CONTRIBUTING.md]: When you are working on your plugin, you can start by writing your unit tests (TDD: test-driven development).

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+We use [Semantic Versioning](https://semver.org/spec/v2.0.0.html) as guideline for the version management.
+
+Steps to release:
+- Create a new branch labeled `release/vX.Y.Z` from the latest `main`.
+- Improve the version number in `package.json`, `package-lock.json` and `changelog.md`.
+- Verify the content of the `changelog.md`.
+- Commit the modifications with the label `Release version X.Y.Z`.
+- Create a pull request on github for this branch into `main`.
+- Once the pull request validated and merged, tag the `main` branch with `vX.Y.Z`
```
Comment (k-y0u) [inline CONTRIBUTING.md]: missing dot

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+- After the tag is pushed, make the release on the tag in GitHub
```
Comment (k-y0u) [inline CONTRIBUTING.md]: missing dot

File: CONTRIBUTING.md (hunk 3 above)
Comment (k-y0u) [inline CONTRIBUTING.md]: The only way to update the `main` branch is through pull request.

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (viennem) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 3 above)
Comment (viennem) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 4 above)
Comment (viennem) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 5 above)
Comment (viennem) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 6 above)
Comment (viennem) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 3 above)
Comment (viennem) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+## Development
+
+### How to test your plugin in leto-modelizer
+
+When you are working on your plugin, you can start by writing your unit tests (TDD: test-driven development).
+
+Alternatively, if you want to test your plugin in leto-modelizer, it can be tedious to rebuild and reinstall your plugin every time you edit its code. Thankfully, there is a trick to auto-reload the Quazar dev server every time you edit your plugin's source code. Here is how to do it:
+
+1. Make sure you cloned the [leto-modelizer](https://github.com/ditrit/leto-modelizer) and your plugin repositories.
+   * You also need to install the dependencies of each cloned repository (by running `cd <repository_path>` and then `npm install`).
```
Comment (viennem) [inline CONTRIBUTING.md:50]: Note: I added this line since the last review.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Zorin95670) [reply CONTRIBUTING.md:62]: or update the `package.json` with this modification

Comment (Zorin95670) [reply CONTRIBUTING.md:62]: See with @viennem, not to be done.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
+# Contributing
```
Comment (Zorin95670) [inline CONTRIBUTING.md:1]: Bad branch name, according to the `Contributing.md` :/

**Pull request closed without merging**